packages/edit-site/src/components/post-edit/index.js

```diff
@@ -42,13 +42,37 @@ function PostEditForm( { postType, postId } ) {
 	);
 	const [ multiEdits, setMultiEdits ] = useState( {} );
 	const { editEntityRecord } = useDispatch( coreDataStore );
-	const { fields } = usePostFields();
+	const { fields: _fields } = usePostFields();
+	const fields = useMemo(
+		() =>
+			_fields?.map( ( field ) => {
+				if ( field.id === 'status' ) {
+					return {
+						...field,
+						elements: field.elements.filter(
+							( element ) => element.value !== 'trash'
+						),
+					};
+				}
+				return field;
+			} ),
+		[ _fields ]
+	);
 	const form = {
 		type: 'panel',
-		fields: [ 'title', 'author', 'date', 'comment_status' ],
+		fields: [ 'title', 'status', 'date', 'author', 'comment_status' ],
 	};
 	const onChange = ( edits ) => {
 		for ( const id of ids ) {
+			if (
```

Comment:
I'm a bit confused by this piece of logic coming from `handleStatus` ([gutenberg/packages/editor/src/components/post-status/index.js](https://github.com/WordPress/gutenberg/blob/b12e89732e467e3bb4f8ebcb296ba224be564131/packages/editor/src/components/post-status/index.js#L150-L164)). Added it in commit https://github.com/WordPress/gutenberg/pull/64398/commits/1babdc0c40386e8a9167c18f3179b83b7ef489e4

My understanding is that by providing a null date, the date component will set it to "right now". If that's correct, shouldn't we use this when `status = publish` and not `future`? As it is, what happens is that the date is reset to "right now" when the user selects `future` (Scheduled) status:

https://github.com/user-attachments/assets/25ad386c-90e1-4254-9651-c63ca67de480

However, in the PostStatus component, the date is reset to "right now" when the user selects "Publish", which makes more sense:

https://github.com/user-attachments/assets/735acad8-12f4-429f-8bae-ec7a0e6fc4bc

If I revert https://github.com/WordPress/gutenberg/commit/1babdc0c40386e8a9167c18f3179b83b7ef489e4 and reset the date when `edits.status=publish` it works as I'd expect.

I'll take a look at this tomorrow.

Reply:
@oandregal The logic in `handleStatus` is checking the previous status and not the one being set. So It's saying like if the previous status was "future", reset the date or something like that.

Reply:
Oh, yes, it's all clear with a fresh brain. Thanks!